text/0000-variadic-generics.md

```diff
@@ -0,0 +1,209 @@
+
+The exact mechanisms used to teach this should be determined after getting more
+experience with how Rustaceans learn. After all, Rust users are a diverse crowd,
+so the "best" way to teach one person might not work as well for another. There
+will need to be some investigation into which explanations are more
+suitable to a general audience.
+
+As for the `(head, ...tail)` joining syntax, this should be explained as
+taking each part of the tail (e.g. `(2, 3, 4)`) and inlining or un-"tupling"
+them (e.g. `2, 3, 4`). This is nicely symmetrical with the `(head, ...tail)`
+pattern-matching syntax.
+
+The `Tuple` trait is a bit of an oddity. It is probably best not to go too
+far into the weeds when explaining it to new users. The extra coherence
+benefits will likely go unnoticed by new users, as they allow for more
+advanced features and wouldn't result in an error where one didn't exist
+before. The obvious exception is when trying to implement the `Tuple` trait.
+Attempts to implement `Tuple` should resort in a relevant error message,
+such as "The `Tuple` trait cannot be implemented for custom types."
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+As with any additions to the language, this RFC would increase the number
+of features present in Rust, potentially resulting increased complexity
+of the language.
+
+There is also some unfortunate overlap between the proposed `(head, ...tail)`
+syntax and the current inclusive range syntax. However, the similarity
+between `start...end` and `...tail` can be disambiguiated by whether or not
+there is an expression immediately before the ellipsis.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Do nothing.
+- Implement one of the other variadic designs, such as
+[#1582](https://github.com/rust-lang/rfcs/pull/1582) or
+[#1921](https://github.com/rust-lang/rfcs/pull/1921)
+- Include explicit `Head`, `Tail`, and `Cons<T>` associated types in the `Tuple`
+trait. This could allow the above syntax to be implemented purely as sugar.
+However, this approach introduces a lot of additional complexity. One of the
+complications is that such a trait couldn't be implemented for `()`, so
+there would have to be separate `Cons` and `Split` traits, rather than one
+unified `Tuple`.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+-It might be useful in the future to expand on the locations where `...Type`
+can be used. Potential extensions to this RFC could allow `...Type` in
+non-tuple generics or in function argument types, like
+`fn foo<Args>(args: ...Args)`.
+This would allow functions and traits to use variadic generics without
+explicit tuples. This could enable things like the proposed `foo[i, j]` syntax
+using`Index<usize, usize>`.
+-Should the `Tuple` trait use separate `TupleRef<'a>` and `TupleMut<'b>` traits
+to avoid dependency on ATCs? It seems nicer to have them all together in one
+trait, but it might not be worth the resulting feature-stacking mess.
+
```

Comment:
`x...` is how you expand a parameter pack in C++, so it'd be familiar to some, for better or worse.